decompiler/config/jak2/ntsc_v1/type_casts.jsonc

```diff
@@ -10237,5 +10245,131 @@
+  "shadow-vu1-add-constants": [
+    [[7, 16], "a2", "dma-packet"],
+    [[20, 66], "v1", "shadow-vu1-constants"],
+    [[72, 77], "a1", "dma-packet"],
+    [[82, 94], "a1", "shadow-vu1-data"]
+  ],
+  "shadow-vu1-add-matrix": [
+    [[11, 19], "a3", "dma-packet"],
+    [[26, 30], "v1", "matrix"]
+  ],
+  "shadow-vu0-upload": [[[16, 18], "a0", "dma-packet"]],
+  "shadow-make-invert-buf": [[[13, 16], "v1", "dma-packet"]],
+  "shadow-invert-z-buf": [
+    [[4, 8], "a2", "dma-packet"],
+    [[13, 16], "a2", "gs-gif-tag"],
+    [27, "t1", "(pointer gs-reg)"],
+    [28, "t1", "(pointer gs-reg64)"],
+    [30, "t1", "(pointer gs-reg)"],
+    [32, "t1", "(pointer gs-frame)"],
+    [34, "t1", "(pointer gs-reg)"],
+    [38, "t1", "(pointer gs-reg)"],
+    [42, "t1", "(pointer gs-reg)"],
+    [36, "t1", "(pointer gs-zbuf)"],
+    [40, "t1", "(pointer gs-test)"],
+    [44, "t1", "(pointer gs-alpha)"],
+    [45, "t1", "(pointer gs-reg)"],
+    [47, "t1", "(pointer gs-reg64)"],
+    [49, "t1", "(pointer gs-reg)"],
+    [51, "t1", "(pointer gs-rgbaq)"],
+    [[61, 64], "t2", "gs-gif-tag"],
+    [89, "t4", "(pointer gs-xyzf)"],
+    [79, "t4", "(pointer gs-xyzf)"],
+    [[106, 117], "v1", "(pointer uint64)"]
+  ],
+  "shadow-dma-init": [
+    [[18, 21], "t4", "dma-packet"],
+    [[27, 31], "t6", "dma-packet"],
+    [[36, 39], "t6", "gs-gif-tag"],
+    [43, "t4", "(pointer gs-reg64)"],
+    [47, "t4", "(pointer gs-test)"],
+    [45, "t4", "(pointer gs-reg64)"],
+    [49, "t4", "(pointer gs-reg64)"],
+    [51, "t4", "(pointer gs-alpha)"],
+    [53, "t4", "(pointer gs-reg64)"],
+    [58, "t4", "(pointer gs-frame)"],
+    [60, "t4", "(pointer gs-reg64)"],
+    [64, "t4", "(pointer gs-reg64)"],
+    [73, "t4", "(pointer gs-reg64)"],
+    [81, "t4", "(pointer gs-reg)"],
+    [84, "t4", "(pointer gs-reg)"],
+    [62, "t4", "(pointer gs-zbuf)"],
+    [71, "t4", "(pointer gs-xy-offset)"],
+    [79, "t4", "(pointer gs-tex0)"],
+    [82, "t4", "(pointer gs-tex1)"],
+    [85, "t4", "(pointer gs-miptbp)"],
+    [87, "t4", "(pointer gs-reg)"],
+    [96, "t4", "(pointer gs-clamp)"],
+    [[103, 106], "t3", "gs-gif-tag"],
+    [[122, 125], "t3", "gs-gif-tag"],
+    [111, "t3", "(pointer gs-prim)"],
+    [113, "t3", "(pointer gs-rgbaq)"],
+    [143, "t5", "(pointer gs-xyzf)"],
+    [156, "t5", "(pointer gs-xyzf)"],
+    [[167, 170], "a3", "gs-gif-tag"],
+    [177, "a3", "(pointer gs-test)"],
+    [179, "a3", "(pointer gs-reg64)"],
+    [183, "a3", "(pointer gs-reg64)"],
+    [193, "a3", "(pointer gs-reg64)"],
+    [190, "a3", "(pointer gs-reg64)"],
+    [202, "a3", "(pointer gs-reg64)"],
+    [181, "a3", "(pointer gs-zbuf)"],
+    [188, "a3", "(pointer gs-frame)"],
+    [191, "a3", "(pointer uint64)"],
+    [206, "a3", "(pointer gs-reg64)"],
+    [213, "a3", "(pointer gs-reg64)"],
+    [214, "a3", "(pointer uint64)"],
+    [216, "a3", "(pointer gs-reg64)"],
+    [211, "a3", "(pointer gs-frame)"],
+    [204, "a3", "(pointer gs-zbuf)"],
+    [200, "a3", "(pointer gs-test)"],
```

Comment:
Can probably get rid of these and trigger a `dma-buffer-add-gs-set` to decompile.